pkg/cli/admin/prune/renderedmachineconfigs/renderedmachineconfigs.go

```diff
@@ -0,0 +1,426 @@
+package renderedmachineconfigs
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"sort"
+	"time"
+	"unicode"
+
+	mcfgclientset "github.com/openshift/client-go/machineconfiguration/clientset/versioned"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/client-go/kubernetes"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+var (
+	pruneMCLong = templates.LongDesc(`
+		# Prune rendered machine configs for an OCP v4 cluster.
+        oc adm prune renderedmachineconfigs
+	`)
+
+	pruneMCExample = templates.Examples(`		
+		# See what the prune command would delete if run with no options
+		oc adm prune renderedmachineconfigs 
+	    # To actually perform the prune operation, the confirm flag must be appended
+		oc adm prune renderedmachineconfigs --confirm
+		# See what the prune command would delete if run on the worker pool
+		oc adm prune renderedmachineconfigs --pool-name=worker
+		# Prunes 10 oldest rendered machine configs in the cluster
+		oc adm prune renderedmachineconfigs --count=10 --confirm
+		# Prunes 10 oldest rendered machine configs in the cluster for the worker pool
+		oc adm prune renderedmachineconfigs --count=10 --pool-name=worker --confirm			
+
+	`)
+
+	pruneMCListLong = templates.LongDesc(`
+		# List rendered machine configs for an OCP v4 cluster.
+        oc adm prune renderedmachineconfigs list
+	`)
+
+	pruneMCListExample = templates.Examples(`
+	    # List all rendered machine configs for the worker pool in the cluster
+		oc adm prune renderedmachineconfigs list --pool-name=worker
+		# List all rendered machine configs in use by the cluster's MachineConfigPools
+		oc adm prune renderedmachineconfigs list --in-use
+
+	`)
+)
+
+type pruneMCOptions struct {
+	RESTClientGetter genericclioptions.RESTClientGetter
+	genericiooptions.IOStreams
+	PoolFilter            string
+	InUse                 bool
+	Confirm               bool
+	Count                 int
+	CurrentRenderedConfig string
+}
+
+func NewCmdPruneMachineConfigs(restClientGetter genericclioptions.RESTClientGetter, streams genericiooptions.IOStreams) *cobra.Command {
+	o := &pruneMCOptions{
+		RESTClientGetter: restClientGetter,
+		IOStreams:        streams,
+		PoolFilter:       "",
+		Count:            0,
+		Confirm:          false,
+	}
+
+	cmd := &cobra.Command{
+		Use:                   "renderedmachineconfigs",
+		DisableFlagsInUseLine: true,
+		Short:                 i18n.T("Prunes machineconfigs in an OpenShift cluster"),
+		Long:                  pruneMCLong,
+		Example:               pruneMCExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Run(context.Background()))
+		},
+	}
+
+	cmd.AddCommand(NewCmdPruneMCList(restClientGetter, streams))
+
+	flags := cmd.Flags()
```

Comment:
It is better to add flags in separate function https://github.com/kubernetes/kubectl/blob/426055bedd6d6e8d1c8ec64a65b0f8048e0c32d1/pkg/cmd/wait/wait.go#L143